src/aggregate/aggregate_request.c

```diff
@@ -1232,13 +1226,12 @@ int buildOutputPipeline(AREQ *req, QueryError *status) {
     RLookup *lookup = AGPLN_GetLookup(pln, NULL, AGPLN_GETLOOKUP_LAST);
     for (size_t ii = 0; ii < req->outFields.numFields; ++ii) {
       ReturnedField *ff = req->outFields.fields + ii;
-      RLookupKey *kk = RLookup_GetKey_TEMP(lookup, ff->name, RLOOKUP_F_NOFLAGS);
+      RLookupKey *kk = RLookup_GetKey(lookup, ff->name, RLOOKUP_M_READ, RLOOKUP_F_NOFLAGS);
       if (!kk) {
         QueryError_SetErrorFmt(status, QUERY_ENOPROPKEY, "No such property `%s`", ff->name);
         goto error;
-      } else if (!(kk->flags & (RLOOKUP_F_SCHEMASRC | RLOOKUP_F_SVSRC))) {
-        // TODO: this is a dead code
-        QueryError_SetErrorFmt(status, QUERY_EINVAL, "Property `%s` is not in document", ff->name);
+      } else if (!(kk->flags & RLOOKUP_F_SCHEMASRC)) {
```

Comment:
Sorting Vector src option is not relevant for READ?

Reply:
It is, but not relevant for this condition. Here we want to validate that the field we requested to highlight is part of the schema. We can actually do the same in the sorter and grouper to preserve the behavior of auto-loading only schema fields, and fail otherwise